ONVIFCamera/Classes/ONVIFCamera.swift

```diff
@@ -167,7 +180,10 @@ public class ONVIFCamera {
     performRequest(request: CameraRequest.getProfiles, response: { (result) in
       guard let body = result["Body"] as? [String: Any],
         let response = body["GetProfilesResponse"]  as? [String: Any],
-        let profilesDict = response["Profiles"] as? [[String: Any]] else { return }
+        let profilesDict = response["Profiles"] as? [[String: Any]] else {
+            profiles([])
+            return
```

Comment:
is there a particular reason for returning an empty array vs return nil?

Reply:
I usually do also. 

However here we can't return nil object unfortunately as "Nil is not compatible with [ONVIFCamera.Profile]. So to make it nilable we need to make  ```@escaping ([Profile]) -> ()) ``` as optional ```@escaping ([Profile]?) -> ()) ```. 

I did not want to change a lot of code especially method declaration. But we can do this and I agree with you on this !

Reply:
Yes let's make it properly! you can go ahead and change the signature of the method and the code calling it in `ViewController` if you want

Reply:
Cool, I just did it :).